src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -683,9 +682,17 @@
              return;
         }
 
-        var objId = X[self.daoKey_].of.ID.type === 'Long' ? parseInt(this.objId) : this.objId;
-
-        return X[this.daoKey_]
+        var objId, daoKey;
+        if ( self.refObj ) {
+          var res = self.refObj.split(':');
```

Comment:
I don't like this. 
what about a 
refDaoKey like causeDaoKey. 
then
refDaoKey
refObjId
and refDaoKey can fallback to daoKey if not provided.

Reply:
sounds good, i will update, thanks